components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -7658,42 +7657,44 @@ public int addComment(APIIdentifier identifier, String commentText, String user)
     /**
      * Adds a comment for an API
      *
-     * @param identifier API Identifier
+     * @param apiTypeWrapper API type wrapper
      * @param comment    Commented Text
      * @param user       User who did the comment
      * @return Comment ID
      */
-    public String addComment(APIIdentifier identifier, Comment comment, String user) throws APIManagementException {
+    public String addComment(ApiTypeWrapper apiTypeWrapper, Comment comment, String user) throws APIManagementException {
         Connection connection = null;
-        ResultSet resultSet = null;
         ResultSet insertSet = null;
-        PreparedStatement getPrepStmt = null;
         PreparedStatement insertPrepStmt = null;
         String commentId = null;
-        int apiId = -1;
+        int id = -1;
+        Boolean isProduct = apiTypeWrapper.isAPIProduct();
 
+        Identifier identifier;
         try {
             connection = APIMgtDBUtil.getConnection();
             connection.setAutoCommit(false);
-
-            String getApiQuery = SQLConstants.GET_API_ID_SQL;
-            getPrepStmt = connection.prepareStatement(getApiQuery);
-            getPrepStmt.setString(1, APIUtil.replaceEmailDomainBack(identifier.getProviderName()));
-            getPrepStmt.setString(2, identifier.getApiName());
-            getPrepStmt.setString(3, identifier.getVersion());
-            resultSet = getPrepStmt.executeQuery();
-            if (resultSet.next()) {
-                apiId = resultSet.getInt("API_ID");
+            //Get API Id
+            if (!isProduct) {
+                identifier = apiTypeWrapper.getApi().getId();
+                id = getAPIID(apiTypeWrapper.getApi().getId(), connection);
+            } else  {
+                identifier = apiTypeWrapper.getApiProduct().getId();
+                id = getAPIID(apiTypeWrapper.getApiProduct().getId(), connection);
```

Comment:
Lets make this,

           ```
 if (!isProduct) {
                identifier = apiTypeWrapper.getApi().getId();
            } else  {
                identifier = apiTypeWrapper.getApiProduct().getId();
           }
     
          id = getAPIID(identifier, connection);
```
Its a lot more cleaner

Reply:
Fixed in 40f428b158055bc3fe7681213df6c8470e1d77a1